File: app/src/main/java/com/chooloo/www/callmanager/ui/recents/RecentsMvpPresenter.java
package com.chooloo.www.callmanager.ui.recents;

import com.chooloo.www.callmanager.entity.RecentCall;
import com.chooloo.www.callmanager.ui.cursor.CursorMvpPresenter;

public interface RecentsMvpPresenter<V extends RecentsMvpView> extends CursorMvpPresenter<V> {
    void onRecentItemClick(RecentCall recentCall);

    boolean onRecentItemLongClick(RecentCall recentCall);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/recents/RecentsMvpPresenter.kt
package com.chooloo.www.callmanager.ui.recents

import com.chooloo.www.callmanager.entity.Recent
import com.chooloo.www.callmanager.ui.list.ListMvpPresenter

interface RecentsMvpPresenter<V : RecentsMvpView?> : ListMvpPresenter<V> {
    fun onRecentItemClick(recent: Recent)
    fun onRecentItemLongClick(recent: Recent): Boolean
}